File: solutions/csharp/cars-assemble/CarsAssemble.cs
static class AssemblyLine
{
    private static double BaseHourlyRate => 221.0;

    public static double SuccessRate(int speed) =>
        speed switch
        {
            0    => 0.00,
            <= 4 => 1.00,
            <= 8 => 0.90,
            9    => 0.80,
            10   => 0.77,
            _    => 0.0
        };
    
    public static double ProductionRatePerHour(int speed) => speed * (BaseHourlyRate * SuccessRate(speed));

    public static int WorkingItemsPerMinute(int speed) => (int) ProductionRatePerHour(speed) / 60;
}

File: solutions/fsharp/cars-assemble/CarsAssemble.fs
module CarsAssemble

let successRate (speed: int): float =
    match speed with
     | 0 -> 0
     | _ when speed < 5 -> 1.00
     | _ when speed < 9 -> 0.90
     | 9 -> 0.8
     | 10 -> 0.77
     | _ -> 0
 
let productionRatePerHour (speed: int): float =
    let baseItemsPerHour = 221 in
    let idealItemsPerHour = float (baseItemsPerHour * speed) in
        (successRate speed) * idealItemsPerHour

let workingItemsPerMinute (speed: int): int =
    let itemsPerHour = productionRatePerHour speed in
        int (itemsPerHour / 60.0)
